src/utils/thread_worker.rs

```diff
@@ -0,0 +1,115 @@
+use skipchannel::*;
+use std::thread::{sleep, spawn, JoinHandle};
+use std::time::Duration;
+
+enum ThreadWorkerMessage<T> {
+    Input(T),
+    StopThread,
+}
+
+pub struct ThreadWorker<Input: Send, Output: Send> {
+    thread: Option<JoinHandle<()>>,
+    to_worker_thread: Sender<ThreadWorkerMessage<Input>>,
+    pub from_worker_thread: Receiver<Output>,
+}
+
+impl<Input: Send, Output: Send> Drop for ThreadWorker<Input, Output> {
+    fn drop(&mut self) {
+        self.to_worker_thread.send(ThreadWorkerMessage::StopThread);
+        let thread = std::mem::replace(&mut self.thread, None);
+        match thread {
+            None => {}
+            Some(join_handle) => {
+                match join_handle.join() {
+                    Ok(()) => {}
+                    Err(error) => eprintln!("{:?}", error),
+                };
+            }
+        }
+    }
+}
+
+impl<Input: Send + 'static, Output: Send + Clone + 'static> ThreadWorker<Input, Output> {
+    pub fn new<F>(mut f: F) -> ThreadWorker<Input, Output>
+    where
+        F: FnMut(Input) -> Output + Send + 'static,
+    {
+        let (to_worker_thread, worker_thread_input) = skipchannel();
+        let (worker_thread_output, from_worker_thread) = skipchannel();
+        let thread = spawn(move || loop {
+            match worker_thread_input.recv() {
+                None => {}
+                Some(ThreadWorkerMessage::Input(input)) => {
+                    worker_thread_output.send(f(input));
+                }
+                Some(ThreadWorkerMessage::StopThread) => break,
+            };
+            sleep(Duration::from_millis(100));
```

Comment:
This uses polling, since `skipchannel::Receiver::recv` doesn't support a blocking method for receiving events. I can't use `std::sync::mpsc::{Sender, Receiver}` instead (which would support a blocking call), because the sending happens in the jack audio thread and `Sender` doesn't implement `Sync`, which is needed for that. (The function that's the problem is this one: https://rustaudio.github.io/rust-jack/jack/struct.Client.html#method.activate_async. That doc link is outdated though, this gained a `Sync` constraint for `ProcessHandler` in a newer version.)